src/rpm.rs

```diff
@@ -44,7 +52,107 @@ impl RPMPackage {
         out.write_all(&self.content)?;
         Ok(())
     }
+
+    // TODO allow passing an external signer/verifier
+
+    /// sign all headers (except for the lead) using an external key and store it as the initial header
+    #[cfg(feature = "signing-meta")]
+    pub fn sign<S>(&mut self, signer: S) -> Result<(), RPMError>
+    where
+        S: crypto::Signing<crypto::algorithm::RSA, Signature = Vec<u8>>
```

Comment:
Can there ever be a `Signature` that is not a byte vec? If so, are we not restricting ourself here by not including a second generic for `Signature`?
On the other hand... if we think it is only a byte vec, we could also remove the Signature type completely.

Reply:
I would not want to restrict external impls too much, as long as it can be accessed as byte slice, things should always be fine. It might be more convenient to have different types being generated for other cases. It just allows more flexibility for the implementer.

Reply:
So the solution would be then to add another generic for the signature type, correct? In its current form, the API does not allow to change the `Signature` type